test/SpokePoolClient.ValidateFill.ts

```diff
@@ -256,62 +254,82 @@ describe("SpokePoolClient: Fill Validation", async function () {
 
-    expect(await spokePoolClient1.binarySearchForBlockContainingDepositId(4)).to.be.greaterThanOrEqual(
-      depositEvents[3].blockNumber
+    expect(await spokePoolClient1.getBlockRangeForDepositId(4)).to.be.greaterThanOrEqual(depositEvents[3].blockNumber);
+    expect(await spokePoolClient1.getBlockRangeForDepositId(3)).to.be.greaterThanOrEqual(depositEvents[2].blockNumber);
+    expect(await spokePoolClient1.getBlockRangeForDepositId(5)).to.be.greaterThanOrEqual(
+      await spokePoolClient1.getBlockRangeForDepositId(4)
     );
-    expect(await spokePoolClient1.binarySearchForBlockContainingDepositId(3)).to.be.greaterThanOrEqual(
-      depositEvents[2].blockNumber
-    );
-    expect(await spokePoolClient1.binarySearchForBlockContainingDepositId(5)).to.be.greaterThanOrEqual(
-      await spokePoolClient1.binarySearchForBlockContainingDepositId(4)
-    );
-    expect(await spokePoolClient1.binarySearchForBlockContainingDepositId(5)).to.be.greaterThanOrEqual(
-      await spokePoolClient1.binarySearchForBlockContainingDepositId(3)
+    expect(await spokePoolClient1.getBlockRangeForDepositId(5)).to.be.greaterThanOrEqual(
+      await spokePoolClient1.getBlockRangeForDepositId(3)
     );
 
     // Last deposit ID is now stricly less than last block in search range because there were more blocks mined since
     // the deposit that incrementerd ID from 4 to 5.
-    expect(await spokePoolClient1.binarySearchForBlockContainingDepositId(5)).to.be.lessThan(
-      spokePoolClient1.latestBlockNumber
-    );
+    expect(await spokePoolClient1.getBlockRangeForDepositId(5)).to.be.lessThan(spokePoolClient1.latestBlockNumber);
 
-    expect(await spokePoolClient1.binarySearchForBlockContainingDepositId(2)).to.be.greaterThanOrEqual(
-      depositEvents[1].blockNumber
-    );
-    expect(await spokePoolClient1.binarySearchForBlockContainingDepositId(2)).to.be.lessThan(
-      depositEvents[3].blockNumber
+    expect(await spokePoolClient1.getBlockRangeForDepositId(2)).to.be.greaterThanOrEqual(depositEvents[1].blockNumber);
+    expect(await spokePoolClient1.getBlockRangeForDepositId(2)).to.be.lessThan(depositEvents[3].blockNumber);
+  });
+
+  it("Fuzz: get search bounds for deposit ID", async function () {
+    const fuzzClient = new MockedSpokePoolClient(
+      createSpyLogger().spyLogger,
+      spokePool_2,
+      null,
+      destinationChainId,
+      spokePool2DeploymentBlock
     );
+
+    const initLow = 0;
+    const initHigh = 1000000;
+    const depositIds = Array(initHigh - initLow + 1).fill(0);
+
+    const testIterations = 100;
+    for (let i = 0; i < testIterations; i++) {
+      // Randomize deposit ID's between initLow and initHigh. ID's should only increase
+      // and will do so 50% of the time. The other 50% of the time they will stay the same.
+      for (let j = 1; j < depositIds.length; j++) {
+        const increment = Math.random() > 0.5 ? 1 : 0;
+        depositIds[j] = depositIds[j - 1] + increment;
+      }
```

Comment:
This is awesome!! However, I think we should consider increments by more than 1, since that's a real edge case that we probably want to handle. Thoughts on this code change to add that?

```suggestion
      for (let j = 1; j < depositIds.length; j++) {
        const increment = Math.max(0, Math.floor((Math.random() - 0.5) * 10));
        depositIds[j] = depositIds[j - 1] + increment;
      }
```

Reply:
Ahh yes this exposed some bugs, nice catch